services/routes-google.js

```diff
@@ -3,55 +3,95 @@ const Toll = require('../models/tolls');
  * return: list of sections
  */
 function findSection (steps, array) {
   const sections = [];
+  const result = [];
+  const path = [];
+  let start;
+  let end;
+
   for (const i in steps) {
+    start = steps[i].start_location;
+    end = steps[i].end_location;
     // checking for the missing tolls
     for (const idx in array) {
-      if ((steps[i].start_location.lat === array[idx].surNorte.lat && steps[i].start_location.lng === array[idx].surNorte.lng) || (steps[i].start_location.lat === array[idx].norteSur.lat && steps[i].start_location.lng === array[idx].norteSur.lng)) {
+      if ((steps[i].start_location.lat === array[idx].surNorte.lat &&
+          steps[i].start_location.lng === array[idx].surNorte.lng) ||
+          (steps[i].start_location.lat === array[idx].norteSur.lat &&
+          steps[i].start_location.lng === array[idx].norteSur.lng)) {
         array[idx].inRoute = true;
       }
     }
     if (steps[i].html_instructions.includes('Toll')) {
       /* iterate throught the result of the road and return it */
       sections.push({
         distance: steps[i].distance.text,
-        start_location: steps[i].start_location,
-        end_location: steps[i].end_location
+        start_location: start,
+        end_location: end
       });
     }
+    path.push({ start, end });
   }
-  return sections;
+
+  return {
+    tolls: sections,
+    missing: result,
+    path: path
+  };
 }
 
 /**
  * Will request to google direction api return a promise
  * return: the Google API data
  */
 const requestRoutesAsync = async (origin, destination) => {
+  if (typeof (origin) === typeof ({})) origin = `${origin.lat},${origin.lng}`;
+  if (typeof (destination) === typeof ({})) destination = `${destination.lat},${destination.lng}`;
+
   const url = 'https://maps.googleapis.com/maps/api/directions/json?' +
-  `origin=${origin.lat},${origin.lng}` +
-  `&destination=${destination.lat},${destination.lng}` +
-  `&key=${keyGoogle}`;
+  `origin=${origin}&destination=${destination}&key=${keyGoogle}`;
 
+  // console.log(url)
   const responseApi = await fetch(url);
-  const responseData = await responseApi.json();
-  if (responseData.status === 'ZERO_RESULTS') return null;
+  // console.log(responseApi)
+  if (responseApi.status === 400) {
+    console.error('Wrong coordinates');
+    return null;
+  }
 
-  return responseData.routes[0].legs[0];
+  let responseData = await responseApi.json();
+  // console.log(responseData)
+  if (responseData.status === 'ZERO_RESULTS' || responseData.status === 'NOT_FOUND') {
+    console.error('Wrong coordinates');
+    return null;
+  }
+
+  responseData = responseData.routes[0].legs[0];
+  if (!validateArea(responseData.start_location) || !validateArea(responseData.end_location)) {
+    console.log(!validateArea(responseData.start_location), !validateArea(responseData.end_location));
+    console.error('The direction is out of the colombia');
+    return null;
+  }
+
+  return responseData;
 };
```

Comment:
## see this
Change the complete function that it can search by name and add conditional to check if the coordinates are out of Colombia